mailpoet/lib/API/JSON/API.php

```diff
@@ -100,13 +93,13 @@ public function setupAjax() {
       $this->setRequestData($_GET, Endpoint::TYPE_GET);
     }
 
-    $ignoreToken = (
-      $this->settings->get('captcha.type') != Captcha::TYPE_DISABLED &&
-      $this->requestEndpoint === 'subscribers' &&
-      $this->requestMethod === 'subscribe'
-    );
+    $uniqueAction = $this->requestData['mailpoet_unique_action'] ?? '';
+    $isTokenValid = $uniqueAction !== ''
+      ? Security::checkTokenForUniqueAction($this->requestToken, $uniqueAction)
+      : Security::checkToken($this->requestToken);
+
 
-    if (!$ignoreToken && $this->checkToken() === false) {
```

Comment:
I couldn't think of a reason we would want to bypass this check, even if it is somewhat redundant. It was adding complexity, and if we can see that a token isn't valid, why would we want to let that request continue?

Reply:
I also have no idea. Found it [here](https://github.com/mailpoet/mailpoet/commit/caa062311221a54a18d3e67b8a06783f97d8c61e) but no reason why it was added. If the captcha works anyway, let's just check it always 👍